src/mempool.rs

```diff
@@ -704,10 +706,22 @@ impl Mempool {
             .iter()
             .map(|s| s.signature.validator.clone())
             .collect();
-        let voting_power = self.staking.compute_voting_power(validators.as_slice());
+        let old_voting_power = self.staking.compute_voting_power(
+            validators
+                .iter()
+                .filter(|v| *v != validator)
+                .cloned()
+                .collect::<Vec<_>>()
+                .as_slice(),
+        );
+        let new_voting_power = self.staking.compute_voting_power(validators.as_slice());
         let f = self.staking.compute_f();
-        // FIXME: This might flood the network with messages for each validator signing after the f+1 signature is received
-        if voting_power > f {
+        // Only send the message if voting power exceeds f, 2*f or 3*f
+        // This garentees that the message is sent only once per threshold
+        if old_voting_power < f && new_voting_power >= f
+            || old_voting_power < 2 * f && new_voting_power >= 2 * f
+            || old_voting_power < 3 * f && new_voting_power >= 3 * f
```

Comment:
I'm pretty sure we assume there is `3*f + 1` staking power total, maybe you could just check for that